configure.py

```diff
@@ -660,8 +660,20 @@ def find_headers(repodir, excluded_dirs):
                 'replica/memtable.cc',
                 'replica/exceptions.cc',
                 'replica/dirty_memory_manager.cc',
+                'mutation/atomic_cell.cc',
+                'mutation/canonical_mutation.cc',
+                'mutation/frozen_mutation.cc',
+                'mutation/mutation.cc',
+                'mutation/mutation_fragment.cc',
+                'mutation/mutation_partition.cc',
+                'mutation/mutation_partition_v2.cc',
+                'mutation/mutation_partition_view.cc',
+                'mutation/mutation_partition_serializer.cc',
+                'mutation/mutation_query.cc',
```

Comment:
The code  in this depends on higher level concepts, like readers, introducing a circular dependency: mutation module --> reader module --> mutation module.
But the dependency already exists, because the mutation fragment knows about the reader_permit, which is arguably part of the reader module. We can sort this out later.

Reply:
I think we'll move out mutation_query to the readers module then. Since it wasn't queued, I'll drop mutation_query.* from the patch.

Reply:
Ah, I see CI failed. Let's hope https://github.com/scylladb/scylladb/commit/fd4ee4878ab6814c5cc119c7538ac0eafa560afb helps.

Reply:
Actually scratch that, `mutation_query.cc` only has `reconcilable_result` in it, so you can include it in here.

Reply:
It does have things like

```
#include "service/priority_manager.hh"
#include "query-result-writer.hh"
```

Reply:
`query-result-writer.hh` is independent of the reader code. It is just a serialization format for mutations, for which the mutation module is the natural place to live.

Reply:
Yes, but does it belong in the mutation module? It might belong in a different layer.

Reply:
query-result-writer.hh, query-result.hh, and mutation_query.cc could be outside the mutation module. They are used to transport query results between replica and coordinator. Mutation module shouldn't need them. So maybe a "query" module. But they're a thin layer, so maybe they can live in the "database" module, which would encompass replica interface.

Reply:
Good idea. We have a lot of files named *query*.hh, I think a query module -- even if thin -- might make sense. But I don't have a strong opinion.
I'm just happy to see files moving into modules, we should do it more.

Reply:
Ok, I'll drop mutation_query.cc out.

What I'm missing is DAG enforcement. We could do it with careful -I flags (so a module only has -I access to to modules it explicitly depends on), but that's sysyphic, or with C++20 modules, but they don't exist.